packages/interface/src/lib.ts

```diff
@@ -95,30 +81,29 @@ export interface UCANOptions {
 }
 
 export interface DelegationOptions<
-  C extends [Capability, ...Capability[]],
+  C extends Capabilities,
   A extends number = number
 > extends UCANOptions {
-  issuer: SigningAuthority<A>
-  audience: Identity
+  issuer: SigningPrincipal<A>
+  audience: Principal
   capabilities: C
+  proofs?: Proof[]
 }
 
-export interface Delegation<
-  C extends [Capability, ...Capability[]] = [Capability, ...Capability[]]
-> {
-  readonly root: Transport.Block<C>
-  readonly blocks: Map<string, Transport.Block>
+export interface Delegation<C extends Capabilities = Capabilities> {
+  readonly root: UCANBlock<C>
+  readonly blocks: Map<string, Block>
```

Comment:
why can't this be UCAN.Block?

Reply:
Because it may contain non UCAN blocks. E.g. when delegation is encoded into a car it may include blocks in capabilities or facts that could be other IPLD things.

Furthermore when we decode invocations from car, we just pass all the blocks into delegation we create even ones that may not be referenced from it and those could include arbitrary things.